src/main/java/randoop/generation/literaltfidf/ScopeToLiteralStatistics.java

```diff
@@ -0,0 +1,152 @@
+package randoop.generation.literaltfidf;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringJoiner;
+import org.checkerframework.checker.nullness.qual.KeyFor;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.plumelib.util.SIList;
+import randoop.main.GenInputsAbstract;
+import randoop.main.RandoopBug;
+import randoop.sequence.Sequence;
+import randoop.types.ClassOrInterfaceType;
+import randoop.types.Type;
+
+/** This class stores information about the literals used in the SUT. */
+public class ScopeToLiteralStatistics {
+
+  /** A special key representing the "all" scope. */
+  public static final Object ALL_SCOPE = "ALL_SCOPE";
+
+  /**
+   * A map from a specific scope to its literal statistics. A null key represents the unnamed
+   * package.
+   */
+  private LinkedHashMap<@Nullable Object, LiteralStatistics> scopeToStatisticsMap =
+      new LinkedHashMap<>();
+
+  /** Creates a ScopeToLiteralStatistics. */
+  public ScopeToLiteralStatistics() {}
+
+  /**
+   * Returns information about literals in a specific scope.
+   *
+   * @param type the type whose scope to access
+   * @return information about literals in the scope for {@code type}, including superclass literals
+   */
+  public LiteralStatistics getLiteralStatistics(ClassOrInterfaceType type) {
+    return scopeToStatisticsMap.computeIfAbsent(getScope(type), __ -> new LiteralStatistics());
+  }
+
+  /**
+   * Returns literal sequences whose output type is exactly {@code neededType}, by searching the
+   * inheritance hierarchy of {@code type} (including all supertypes). Designed for CLASS-level
+   * literal selection.
+   *
+   * @param type the starting type whose inheritance hierarchy determines which scopes to search
```

Comment:
For clarity and for parallelism with `neededType`, how about renaming `type` to `definingType`?